sunpy/util/util.py

```diff
@@ -195,3 +197,54 @@ def hash_file(path):
             sha256.update(data)
 
     return sha256.hexdigest()
+
+
+def get_rectangle_coordinates(bottom_left, *, top_right = None, width: u.deg = None, height: u.deg = None):
+    """
+    Returns bottom_left and top_right coordinates of a 2D rectangular region of interest.
+
+    Parameters
+    ----------
+        bottom_left : `astropy.coordinates.BaseCoordinateFrame` or `~astropy.coordinates.SkyCoord`
+            The bottom_left coordinate of the rectangle. If a `SkyCoord` it can
+            have shape ``(2,)`` and also define ``top_right``.
+
+        top_right : `astropy.coordinates.BaseCoordinateFrame` or `~astropy.coordinates.SkyCoord`
+            The top_right coordinate of the rectangle. Can only be omitted if
+            ``bottom_left`` has shape ``(2,)``.
+
+        width : `astropy.units.Quantity`
+            The width of the rectangle.
+            Must be omitted if ``bottom_left`` has shape ``(2,)`` or ``top_right`` is provided.
+
+        height : `astropy.units.Quantity`
+            The height of the rectangle.
+            Must be omitted if ``bottom_left`` has shape ``(2,)`` or ``top_right`` is provided.
+
+    """
+    if ((all((width, height)) and (top_right is not None or bottom_left.shape == (2,))) or
```

Comment:
For the sake of both readability of this if statement :grinning: and the specificity of the error I suggest you break this up into one if statement which raises an error "if bottom_left and top_right specified width and height must be None" and one which says the opposite.